action/protocol/staking/handlers.go

```diff
@@ -699,12 +699,12 @@ func (p *Protocol) handleCandidateRegister(ctx context.Context, act *action.Cand
 			}
 		}
 		candID = id
-	}
-	c = csm.GetByIdentifier(candID)
-	if c != nil {
-		return log, nil, &handleError{
-			err:           ErrInvalidOwner,
-			failureStatus: iotextypes.ReceiptStatus_ErrCandidateAlreadyExist,
+		c = csm.GetByIdentifier(candID)
```

Comment:
should we overwrite `c` or use a different var name?
`c` is used at L711 and below, should `c` be the value of GetByOwner or the value of GetByIdentifier()?

Reply:
i mean, this would fix the bug, but it affects the logic/meaning of L711 (and below) after hard-fork